### PR TITLE
Renaming to music_player and moving from strands_deployment to here

### PR DESCRIPTION
**Pull request by cdondrup**
Needs https://github.com/strands-project/strands_deployment/pull/81 merged.

### CODE REVIEW

File: music_player/setup.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+
+from distutils.core import setup
+from catkin_pkg.python_setup import generate_distutils_setup
+
+d = generate_distutils_setup(
+   #  don't do this unless you want a globally visible script
+   # scripts=['bin/myscript'], 
+   packages=['music_player'],
```
Comment (marc-hanheide) [inline music_player/setup.py]: the folder in src is still called `aaf_music_player`???

File: music_player/src/aaf_music_player/aaf_music_player_server.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
```
Comment (marc-hanheide) [inline music_player/src/aaf_music_player/aaf_music_player_server.py]: I think the file and dir names are incorrect

**Comment by marc-hanheide** [comment]
The player node is still called `aaf_music_player_server.py`, and it should probably by moved to `scripts/`?

File: music_player/setup.py [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/usr/bin/env python
+
+from distutils.core import setup
+from catkin_pkg.python_setup import generate_distutils_setup
+
+d = generate_distutils_setup(
+   #  don't do this unless you want a globally visible script
+   # scripts=['bin/myscript'],
```
Comment (marc-hanheide) [inline music_player/setup.py]: the server node `aaf_music_player_server.py` should be installed here! (and renamed to `music_player_server.py`)

Comment (marc-hanheide) [inline music_player/setup.py]: and as I said it should be in `scripts/`

**Comment by cdondrup** [comment]
Yes, by coincidence I just figured the same after desperately trying to install it.

**Comment by cdondrup** [comment]
Is there any benefit in using the setup.py over a install target in the CMakeLists.txt?

**Comment by marc-hanheide** [comment]
it's an easier and more "pythonistic" way

**Comment by nilsbore** [comment]
I did not know what the difference was between putting it in scripts and src. Is it good practice to put it in scripts?

**Comment by marc-hanheide** [comment]
everything that is to be run with rosrun or roslaunch should be in scripts

**Comment by marc-hanheide** [comment]
also, the `__init__.py` looks like this:

```
import music_player_server

__all__ = ['music_player_server']
```

which doesn't work... so I suggest to `mv music_player/src/music_player/aaf_music_player_server.py music_player/scripts/music_player_server.py` and get rid of `src/` altogether.

**Comment by cdondrup** [comment]
Just found this quote on http://docs.ros.org/indigo/api/catkin/html/user_guide/setup_dot_py.html

"ROS Users should generally not use the scripts argument, as in ROS, executables should be executed using rosrun rather than being installed to the global bin folder. One way of installing such python scripts is to add the following to the CMakeLists.txt: ..."

I knew there was a reason why I never used the setup.py to install executables. `rosrun` just cannot find them afterwards.

I will set an install target in the CMakeLists.txt and be done with it. All the renaming and moving is done already, just didn't push it yet. I'll fix it, just give me a minute.

@nilsbore executables must be installed via the `CMakeLists.txt` to keep their permissions and be put into the right location. If you install something via the `setup.py` it is either put in the wrong location if you use the `scripts` arguments, or the executable rights are revoked if you use the `packages` argument. The `packages` argument is just meant for non executable source files. The scripts by convention should be in the `scripts` directory but don't have to. Also, as @marc-hanheide said, the `__init__.py` should just be empty, as it filled by ROS automatically.

**Comment by marc-hanheide** [comment]
yes, you are right as always

**Comment by cdondrup** [comment]
This works for me now and @Jailander said we have to stop spamming his email account ;)
@nilsbore can you check the aaf_waypoint_sounds installs correctly when you base it on the pygame_managed_player?

**Comment by nilsbore** [comment]
@cdondrup Thanks for the info! I will make sure when I check the other one!

**Comment by marc-hanheide** [comment]
I'm happy now :-)

**Comment by marc-hanheide** [comment]
I trigger a new release strands_ui so that other packages can use the music player and the fixed mary

**Pull request merged**